src/v0/destinations/intercom/transform.js

```diff
@@ -89,30 +90,72 @@ function validateTrack(message, payload) {
   throw new InstrumentationError('Email or userId is mandatory');
 }
 
+function attachUserAndCompany(message, Config) {
+  if (!message.userId && !message.email) {
+    return;
+  }
+  const requestBody = {};
+  if (message.userId) {
+    requestBody.user_id = message.userId;
+  }
+  if (message.email) {
+    requestBody.email = message.email;
+  }
+  const companyObj = {};
+  companyObj.company_id = message.groupId;
+  companyObj.name = message.name;
+  requestBody.companies = [companyObj];
+  let response = defaultRequestConfig();
+  response.method = defaultPostRequestConfig.requestMethod;
+  response.endpoint = ConfigCategory.IDENTIFY.endpoint;
+  response.headers = {
+    'Content-Type': 'application/json',
+    Authorization: `Bearer ${Config.apiKey}`,
+    Accept: 'application/json',
+    'Intercom-Version': '1.4',
+  };
+  response.body.JSON = requestBody;
+  return response;
+};
+
 function validateAndBuildResponse(message, payload, category, destination) {
+  const respList = [];
+  let response = defaultRequestConfig();
+  response.method = defaultPostRequestConfig.requestMethod;
+  response.endpoint = category.endpoint;
+  response.headers = {
+    'Content-Type': 'application/json',
+    Authorization: `Bearer ${destination.Config.apiKey}`,
+    Accept: 'application/json',
+    'Intercom-Version': '1.4',
+  };
+  response.userId = message.anonymousId;
   const messageType = message.type.toLowerCase();
-  const response = defaultRequestConfig();
   switch (messageType) {
     case EventType.IDENTIFY:
       response.body.JSON = removeUndefinedAndNullValues(validateIdentify(message, payload));
       break;
     case EventType.TRACK:
       response.body.JSON = removeUndefinedAndNullValues(validateTrack(message, payload));
       break;
+    case EventType.GROUP:
+      response.body.JSON = removeUndefinedAndNullValues(payload);
+      respList.push(response);
+      try {
+        response = attachUserAndCompany(message, destination.Config);
+        if (response) {
+          response.userId = message.anonymousId;
+          respList.push(response);
+        }
+      } catch (exp) {
+        logger.info("failed to send /users call in group event", exp);
```

Comment:
Do we need this?

Reply:
I added this log to to catch any discrepency when linking user and company